### PR TITLE
Preprocess-part2

### PR DESCRIPTION
**Pull request by josephignace**
Stage1 data cleaning.

### CODE REVIEW

File: analysis/preprocess/Stage1_data_cleaning.R [hunk 1]
```diff
@@ -0,0 +1,374 @@
...
+    
+    print(paste0(cohort_name, " " , nrow(input), " rows in the input file after cohort specificinclusion criteria"))
+    
+    
+    
+    print(paste0(cohort_name, " ", nrow(input), " rows in the input file after common inclusion criteria"))
+    #-------------------------------------------------#
+    # 3.c. Apply outcome specific exclusions criteria
+    #-------------------------------------------------#
+     ####INSERT OUTCOME SPECIFIC EXCLUSION HERE#######
```
Comment (marwaalarab) [inline analysis/preprocess/Stage1_data_cleaning.R:325]: You need to apply MH outcomes exclusions here if any.

File: analysis/study_definition_prevax.py [hunk 2]
```diff
@@ -79,6 +79,13 @@
     #     returning = 'death_date',
     #     returning_type = 'date', 
     # ),
+    
+    ## Any covid vaccination, identified by target disease
+    vax_date_covid_1 = patients.with_value_from_file(
+        f_path = 'output/index_dates.csv',
+        returning = 'vax_date_covid_1',
+        returning_type = 'date'          
+    ),
```
Comment (marwaalarab) [inline analysis/study_definition_prevax.py:89]: do we need vax_date_covid_1 in prevax ?

**Review by marwaalarab — COMMENTED** [review]
Hi Jose, 
All looks alright. Just 2 comments which won't necessarily  require any change. I will approve after your reply.

File: analysis/study_definition_prevax.py (hunk 2 above)
Comment (josephignace) [reply analysis/study_definition_prevax.py:89]: It won't run preprocess without it. I checked the old repo approach, and vax_date_covid_1 is there. I also checked diabetes repo and it is there. I would suggest to talk about it with Venexia to confirm.

File: analysis/preprocess/Stage1_data_cleaning.R (hunk 1 above)
Comment (josephignace) [reply analysis/preprocess/Stage1_data_cleaning.R:325]: Hi Marwa,
There are not exclusion criteria for MH.

**Pull request merged**